src/sql/workbench/services/blob/browser/blobService.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as mssql from 'mssql';
+import { IAzureBlobService } from 'sql/platform/blob/common/blobService';
+import { MainThreadBlob } from 'sql/workbench/api/browser/mainThreadAzureBlob';
+
+export class AzureBlobService implements IAzureBlobService {
+
+	public _serviceBrand: undefined;
+	private _proxy: MainThreadBlob;
+
+	/**
+	 * Internal use only, do not call! This is called once on startup by the proxy object used
+	 * to communicate with the extension host once it's been created.
+	 * @param proxy The proxy to use to communicate with the azurecore extension
+	 */
+	public registerProxy(proxy: MainThreadBlob) {
+		this._proxy = proxy;
+	}
+
+	public createSas(connectionUri: string, blobContainerUri: string, blobContainerKey: string, storageAccountName: string, expirationDate: string): Thenable<mssql.CreateSasResponse> {
```

Comment:
Internal stuff should use Promises (you can await the proxy call and return that to turn it into a promise)